### ISSUE TITLE
'for' loops without braces are handled incorrectly

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2677]
```r
lintr::lint("
for (i in 1:10)
  for (j in 1:10)
    i + j
", linters = indentation_linter())
# <text>:3:2: style: [indentation_linter] Indentation should be 0 spaces but is 2 spaces.
#   for (j in 1:10)
# ~^
# <text>:4:4: style: [indentation_linter] Indentation should be 0 spaces but is 4 spaces.
#     i + j
# ~~~^
```

Came across this today. Here, braces are not strictly required. The style guide requires them, but not the R parser.

Therefore I'd expect the above to produce no lints, but we get 2.

**Comment by MichaelChirico** [comment]
Actually I think it's nothing to do with the nested `for`, but rather any `for` loop without `{` enclosing the body has this issue:

```r
lint("for (i in 1:10)\n  i", indentation_linter())
# <text>:2:2: style: [indentation_linter] Indentation should be 0 spaces but is 2 spaces.
#   i
# ~^
```

**Comment by MichaelChirico** [comment]
Oh, this is #2564
